cmd/auth-rpc-client.go

```diff
@@ -80,13 +80,13 @@ func newAuthRPCClient(config authConfig) *AuthRPCClient {
 
 // Login - a jwt based authentication is performed with rpc server.
 func (authClient *AuthRPCClient) Login() (err error) {
-	authClient.Lock()
-	defer authClient.Unlock()
-
+	authClient.RLock()
 	// Return if already logged in.
 	if authClient.authToken != "" {
+		authClient.RUnlock()
 		return nil
 	}
+	authClient.RUnlock()
```

Comment:
I understood `RLock()` is right thing for `call()`, but for `Login()` the whole method needs to be protected by `Lock()`